runtime/lua/vim/ui.lua

```diff
@@ -104,4 +104,45 @@ function M.input(opts, on_confirm)
   end
 end
 
+--- Opens a URI by sending it to the system's default handler.
+--- This function utilizes `xdg-open`, `wslview`, or `open` commands depending on
+--- the operating system to open the provided URI.
+---
+---
+---@param uri string Uri to be opened
+---   TODO: uri should accept vim.uri.Uri
+---
+--- Example:
+--- <pre>lua
+--- vim.ui.open_uri("https://neovim.io/")
+---
+--- vim.ui.open_uri("/path/to/file")
+--- </pre>
+function M.open_uri(uri)
+  local cmd
+
+  local sysname = vim.loop.os_uname().sysname
+
+  if sysname == 'Darwin' then
+    cmd = 'open'
+  elseif sysname:match('windows') then
+    cmd = 'wslview'
+  else -- Assume linux
+    cmd = 'xdg-open'
+  end
+
+  local ret = vim.fn.jobstart({
```

Comment:
Could this return the `ret` and maybe have a second `opts` parameter with `quiet` to surpress the message? Then it could be used in https://github.com/neovim/neovim/blob/02d5a678fbf4ed36b4896010a051f136d587e779/runtime/lua/vim/lsp/handlers.lua#L571-L580

Reply:
It might also be good to have a confirmation for security.

Reply:
> I'm not sure about the opts/quiet, maybe it's fine to always have the error notification.

`system()` and `jobstart()` will complain either way.